sql/_13_issues/_24_2h/cases/cbrd_25316_2.sql

```diff
@@ -0,0 +1,159 @@
+ *   | COUNT '(' '*' ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | COUNT '(' of_distinct_unique expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | COUNT '(' opt_all expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic '(' of_distinct_unique expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic '(' opt_all expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_first_last '(' expression_ ')' opt_analytic_ignore_nulls OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_nth_value '(' expression_ ',' expression_ ')' opt_analytic_from_last opt_analytic_ignore_nulls OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_lead_lag '(' expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_lead_lag '(' expression_ ',' expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_lead_lag '(' expression_ ',' expression_ ',' expression_ ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_analytic_no_args '(' ')' OVER '(' opt_analytic_partition_by opt_analytic_order_by ')'
+ *   | of_percentile '(' expression_ ')' WITHIN GROUP_ '(' ORDER BY sort_spec ')' opt_over_analytic_partition_by
+ *
+ * of_analytic
+ *   : avg
+ *   | max
+ *   | min
+ *   | sum
+ *   | stddev
+ *   | stddev_pop
+ *   | stddev_samp
+ *   | var_pop
+ *   | var_samp
+ *   | variance
+ *   | ntile
+ *   | median
+ *
+ * of_analytic_first_last
+ *   : FIRST_VALUE
+ *   | LAST_VALUE
+ *
+ * of_analytic_nth_value
+ *   : NTH_VALUE
+ *
+ * of_analytic_lead_lag
+ *   : LEAD
+ *   | LAG
+ *
+ * of_analytic_no_args
+ *   : ROW_NUMBER
+ *   | RANK
+ *   | DENSE_RANK
+ *   | CUME_DIST
+ *   | PERCENT_RANK
+ *
+ * of_percentile
+ *   : PERCENTILE_CONT
+ *   | PERCENTILE_DISC
+ *
+ * ---------------------------------------- */
+
+select '' as "test data";
+drop table if exists tbl_a, tbl_b;
+create table tbl_a (col_a int, c_r float, c_r_nl float, c_r_merge float, index idx_a (col_a));
+create table tbl_b (col_a int, col_b int, c_r float, index idx_b (col_b));
+
+insert into tbl_a
+with recursive cte (n) as (
+    select 1
+    union all
+    select n + 1 from cte where n < 10
+  )
+select n, null, null, null from cte;
+insert into tbl_b select a.col_a, b.col_a, null from tbl_a a, tbl_a b where b.col_a <= a.col_a;
+
+update statistics on tbl_a with fullscan;
+update statistics on tbl_b with fullscan;
+
+
+select 'The UPDATE JOIN statement does not allow the use of analytic functions.' as "test case 2";
+
+select '' as "test case2: use avg";
+update /*+ recompile */ tbl_a a, tbl_b b set a.c_r = avg(b.col_b) over (partition by b.col_a) where a.col_a = b.col_a;
+select col_a, to_char(c_r), if(isnull(c_r)=1, 'pass', 'fail') from tbl_a order by col_a;
```

Comment:
여기서는 에러가 발생여부만 확인해도 된다고 생각합니다.
```suggestion
update /*+ recompile */ tbl_a a, tbl_b b set a.c_r = avg(b.col_b) over (partition by b.col_a) where a.col_a = b.col_a;
```

Reply:
리뷰 반영되었습니다.